internal/okta/server/handler.go

```diff
@@ -0,0 +1,300 @@
+	SignOnMode  string           `json:"signOnMode"`
+	Credentials *OktaCredentials `json:"credentials,omitempty"`
+	Settings    *OktaSettings    `json:"settings,omitempty"`
+}
+
+type OktaCredentials struct {
+	OAuthClient *OktaOAuthClient `json:"oauthClient,omitempty"`
+}
+
+type OktaOAuthClient struct {
+	ClientId                string `json:"client_id,omitempty"`
+	ClientSecret            string `json:"client_secret,omitempty"`
+	TokenEndpointAuthMethod string `json:"token_endpoint_auth_method,omitempty"`
+	ClientUri               string `json:"client_uri,omitempty"`
+}
+
+type OktaSettings struct {
+	OAuthClient *OktaOAuthClientSettings `json:"oauthClient,omitempty"`
+}
+
+type OktaOAuthClientSettings struct {
+	ClientUri       string   `json:"client_uri,omitempty"`
+	LogoUri         string   `json:"logo_uri,omitempty"`
+	RedirectUris    []string `json:"redirect_uris,omitempty"`
+	ResponseTypes   []string `json:"response_types,omitempty"`
+	GrantTypes      []string `json:"grant_types,omitempty"`
+	ApplicationType string   `json:"application_type,omitempty"`
+	ConsentMethod   string   `json:"consent_method,omitempty"`
+	IssuerMode      string   `json:"issuer_mode,omitempty"`
+}
+
+type OktaError struct {
+	ErrorCode    string `json:"errorCode"`
+	ErrorSummary string `json:"errorSummary"`
+	ErrorLink    string `json:"errorLink,omitempty"`
+	ErrorId      string `json:"errorId,omitempty"`
+}
+
+func NewStrictServerHandler(opts *Options, restyClient *resty.Client) *StrictServerHandler {
+	// Set up authentication for all requests
+	restyClient.OnBeforeRequest(func(c *resty.Client, r *resty.Request) error {
+		r.SetHeader("Authorization", "SSWS "+opts.APIToken)
+		r.SetHeader("Accept", "application/json")
+		r.SetHeader("Content-Type", "application/json")
+		r.SetError(&OktaError{})
+		return nil
+	})
+
+	return &StrictServerHandler{
+		restClient: restyClient,
+		oktaDomain: opts.OktaDomain,
+		apiToken:   opts.APIToken,
+	}
+}
+
+// CreateOAuthApplication creates a client in Okta
+func (s *StrictServerHandler) CreateOAuthApplication(
+	_ context.Context,
+	request portalv1.CreateOAuthApplicationRequestObject,
+) (portalv1.CreateOAuthApplicationResponseObject, error) {
+	if request.Body == nil || len(request.Body.Id) == 0 {
+		return portalv1.CreateOAuthApplication400JSONResponse(newPortal400Error("unique id is required")), nil
+	}
+
+	// Create OAuth 2.0 Service Application in Okta
+	app := OktaApplication{
+		Name:       "oidc_client", // Use correct Okta application name
+		Label:      request.Body.Id,
+		SignOnMode: "OPENID_CONNECT",
+		Credentials: &OktaCredentials{
+			OAuthClient: &OktaOAuthClient{
+				TokenEndpointAuthMethod: "client_secret_basic",
+			},
+		},
+		Settings: &OktaSettings{
+			OAuthClient: &OktaOAuthClientSettings{
+				GrantTypes:      []string{"client_credentials"},
+				ApplicationType: "service",
+				ConsentMethod:   "TRUSTED",
+				IssuerMode:      "ORG_URL",
+			},
```

Comment:
Looking at the API [here](https://developer.okta.com/docs/api/openapi/okta-management/management/tag/Application/#tag/Application/operation/createApplication!path=0/settings&t=request) these fields seem to be a bit off, it looks like only the login url needs to be specified.